File: abp/Arborizavel.java
public interface Arborizavel<T> {

    NoTriplo<T> getRaiz();
    void inserir(T dado);
    NoTriplo<T> apagar(NoTriplo<T> raiz, T dado);
    boolean existe(T dado);
    NoTriplo<T> buscar(T dado);     
    String imprimirPreOrdem(NoTriplo<T> raiz);
    String imprimirEmOrdem(NoTriplo<T> raiz);
    String imprimirPosOrdem(NoTriplo<T> raiz);
}

File: arvore/Kotlin/abp/abp/Arborizavel.kt
interface Arborizavel {
    fun getRaiz(): NoTriplo?
    fun inserir(dado: Any?)
    fun apagar(dado: Any?): Boolean
    fun existe(dado: Any?): Boolean
    fun buscar(dado: Any): NoTriplo?
    fun limpar()
    fun imprimirPreOrdem(): String
    fun imprimirEmOrdem(): String
    fun imprimirPosOrdem(): String
}